src/Domain/FunctionsOfTime/QuaternionFunctionOfTime.cpp

```diff
@@ -0,0 +1,213 @@
+         "Currently there are "
+             << omega_deriv_info.size() << " stored omegas and "
+             << stored_quaternions_and_times_.size()
+             << " stored quaternions. There must be at least the same number "
+                "of stored quaternions as omegas otherwise we cannot solve the "
+                "ode and populate the missing quaternions.");
+
+  // If they are already the same size, we don't have to update anything so exit
+  // now
+  if (omega_deriv_info.size() == stored_quaternions_and_times_.size()) {
+    return;
+  }
+
+  // Will store the solution of the ode
+  boost::math::quaternion<double> quaternion_to_integrate;
+  for (size_t i = stored_quaternions_and_times_.size();
+       i < omega_deriv_info.size(); i++) {
+    // Final time, initial time, and quaternion
+    const double t = omega_deriv_info[i].time;
+    const double t0 = omega_deriv_info[i - 1].time;
+    quaternion_to_integrate = datavector_to_quaternion(
+        stored_quaternions_and_times_[i - 1].stored_quantities[0]);
+
+    solve_quaternion_ode(make_not_null(&quaternion_to_integrate), t0, t);
+
+    // normalize quaternion
+    normalize_quaternion(make_not_null(&quaternion_to_integrate));
+
+    stored_quaternions_and_times_.emplace_back(
+        t, std::array<DataVector, 1>{
+               quaternion_to_datavector(quaternion_to_integrate)});
+  }
+
+  ASSERT(omega_deriv_info.size() == stored_quaternions_and_times_.size(),
+         "The number of stored omegas must be the same as the number of stored "
+         "quaternions after updating the missing quaternions. Now there are "
+             << omega_deriv_info.size() << " stored omegas and "
+             << stored_quaternions_and_times_.size() << " stored quaternions.");
+}
+
+template <size_t MaxDeriv>
+void QuaternionFunctionOfTime<MaxDeriv>::solve_quaternion_ode(
+    const gsl::not_null<boost::math::quaternion<double>*>
+        quaternion_to_integrate,
+    const double t0, const double t) const noexcept {
+  // lambda that stores the internals of the ode
+  const auto quaternion_ode_system =
+      [this](const boost::math::quaternion<double>& state,
+             boost::math::quaternion<double>& dt_state,
+             const double time) noexcept {
+        const boost::math::quaternion<double> omega =
+            datavector_to_quaternion(omega_f_of_t_.func(time)[0]);
+        dt_state = 0.5 * state * omega;
+      };
+
+  // Dense stepper
+  auto dense_stepper = boost::numeric::odeint::make_dense_output(
+      1.0e-15, 1.0e-14,
+      boost::numeric::odeint::runge_kutta_dopri5<
+          boost::math::quaternion<double>, double,
+          boost::math::quaternion<double>, double,
+          boost::numeric::odeint::vector_space_algebra>{});
+
+  // Integrate from t0 to t, storing result in quaternion_to_integrate
+  boost::numeric::odeint::integrate_adaptive(
+      dense_stepper, quaternion_ode_system, *quaternion_to_integrate, t0, t,
+      1e-4);
```

Comment:
In it's infinite wisdom, GitHub has apparently decided I can't reply to the old thread, so, new continuation thread.

Perhaps a better way to express my concern here is to point out that this number isn't dimensionless.

Anyway, though, I've thought about this and this value is probably fine for any case we will hit.  The required value probably scales with 1/omega, and omega can't really exceed 1/(total mass).  We always choose the total mass to be 1 or use physical values of >1 solar mass.  Your test uses omega > 1, so that should be more restrictive.